spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/spanner/GcpSpannerAutoConfiguration.java

```diff
@@ -123,7 +124,8 @@
 
 		@Bean
 		@ConditionalOnMissingBean
-		public SpannerOptions spannerOptions(SessionPoolOptions sessionPoolOptions) {
+		public SpannerOptions spannerOptions(SessionPoolOptions sessionPoolOptions,
+											Optional<SpannerOptionsCustomizer> customizer) {
 			Builder builder = SpannerOptions.newBuilder()
```

Comment:
Do you want to make this builder a bean?

Reply:
Sure. With all the configured properties, right? And then the `SpannerOptions` bean will only apply customizer and build.

Reply:
Why not apply the customizer in the builder bean? What does Spring do in such cases?